src/kernels/jupyter/launcher/liveshare/hostJupyterServer.ts

```diff
@@ -254,9 +254,9 @@ export class HostJupyterServer implements INotebookServer {
     }
 
     // Return a copy of the connection information that this server used to connect with
-    public getConnectionInfo(): IJupyterConnection | undefined {
+    public getConnectionInfo(): IJupyterConnection {
         if (!this.connection) {
-            return undefined;
+            throw new Error('Not connected');
```

Comment:
THis is impossible, when the class is contructed, the first things that called is the `connect` method that initializes the property `this.connection`. Ideally we shoudl remove such code that initializes the class and then calls methods.
I've found this in a number of places and replaced them with factories, after that we dont end up with nullables with complicates the code.

Reply:
Will change to a factory in another PR